api/src/Entity/CopyFromPrototype.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Entity;
+
+use App\Util\EntityMap;
+
+class CopyFromPrototype {
+    /**
+     * @param BaseEntity $entity
+     * @param BaseEntity $prototype
+     * @param EntityMap  $entityMap
+     */
+    public static function add($entity, $prototype, &$entityMap = null) {
```

Comment:
Cool dass du das extrahieren konntest. Kannst du mich nochmals erinnern warum es nötig ist, die EntityMap anfangs als `null` hineinzugeben, mit pass-by-reference? Warum können wir nicht einfach beim ersten Aufruf von `copyFromPrototype` (aktuell also im CampDataPersister) eine neue EntityMap erstellen und hineingeben? Dann würde sich die Zeile
```php
CopyFromPrototype::add($this, $prototype, $entityMap);
```
jeweils vereinfachen auf
```php
$entityMap->add($prototype, $entity);
```

Reply:
Dein Vorschlag würde auch funktionieren.
Der Gedanke war - als Anwender musst du nichts weiter tun, als `copyFromPrototype($prototype)` aufrufen. Du musst dich um keine Details kümmern.

Reply:
Ja, sehe ich. Dieser Wunsch ist wohl ein Stück weit eine Folge davon, dass wir jetzt diese Business Logik auf die Entities packen, statt in Services.
Im DataPersister könnten wir aber möglicherweise sogar DI verwenden um jedes Mal eine neue EntityMap zu [konstruieren](https://symfony.com/doc/current/service_container/shared.html). Dann wäre der IriConverter auch gratis, und es wäre vermutlich etwas einfacher testbar. Reden wir doch am Meeting nochmals darüber.

Reply:
CopyFromPrototype::add
Ebenfalls ref $entitymap = null entfernen